src/Bones/bone.ts

```diff
@@ -71,8 +70,15 @@ export class Bone extends Node {
 
     /** @hidden */
     set _matrix(value: Matrix) {
+        this._needToCompose = false; // in case there was a pending compose

```

Comment:
I would have moved this after the check on the `updateFlag`: if the `_needToCompose` flag is `true` when calling the `_matrix` setter it means the current local matrix needs to be composed and we should keep the flag `true` if we early return because the matrices are identical.

Reply:
No, that's not right. If this flag is true, that means someone set the bone transform using TRS. If that is still in progress and someone sets the matrix, then we want to ignore the TRS.